pkg/controllers/resources/nodes/syncer.go

```diff
@@ -291,7 +291,10 @@ func (s *nodeSyncer) shouldSync(ctx context.Context, pObj *corev1.Node) (bool, e
 			ls = labels.Set{}
 		}
 
-		return s.nodeSelector.Matches(ls), nil
+		if !s.nodeSelector.Matches(ls) {
```

Comment:
The value of [EnforceNodeSelector option (--enforce-node-selector)](https://github.com/loft-sh/vcluster/blob/74ac1af61f884dc46ef14e66f1b203c05ed80aeb/cmd/vcluster/cmd/start.go#L116) should be checked too, we should allow nodes that don't match the selector unless this option is explicitly set to false.

Reply:
Thanks for the feedback, making those changes.

Reply:
I agree this should be done.
 
There are only two scenarios I can see where a node would end up being included where it shouldn't with `--enforce-node-selector`:
- Syncer was started without `--enforce-node-selector`, or a different set of labels to match on, and then alter restarted with another set of labels while there are already pods running in the cluter.
- A node was re-labeled.

Reply:
@salanki yes, you described the edge cases well. I don't this we need to solve these, so the current implementation is good! :)